packages/hardhat-core/src/internal/hardhat-network/provider/modules/eth.ts

```diff
@@ -1189,11 +1202,15 @@ export class EthModule {
       return tag.toString();
     }
 
-    return bufferToHex(tag);
+    if ("blockNumber" in tag) {
+      return tag.blockNumber.toString();
+    }
+
+    return bufferToHex(tag.blockHash);
   }
 
   private _extractNormalizedLogTopics(
-    topics: LogTopics
+    topics: OptionalRpcLogTopics
```

Comment:
This change from non optional to optional may look weird. It's because `logTopic` and `logAddress` already included `undefined`. I changed them to match what the rest of the types do.